wsmaster/che-core-api-factory-github/src/main/java/org/eclipse/che/api/factory/server/github/GithubURLParser.java

```diff
@@ -43,6 +43,10 @@
           ".*<div class=\"State[\\s|\\S]+(?<prState>Closed|Open|Merged)[\\s|\\S]+<\\/div>[\\s|\\S]+into[\\s]+(from[\\s]*)*<span title=\"(?<prRepoUser>[^\\\\/]+)\\/(?<prRepoName>[^\\:]+):(?<prBranch>[^\\\"]+).*",
           Pattern.DOTALL);
 
+  /** Regexp to check if repository name contains .git extension and make matching group w/out it */
+  protected static final Pattern GIT_EXTENSION_PATTERN =
+      Pattern.compile("^([\\S]+(?=\\.git))(?:\\.git|(/))?$");
```

Comment:
this code do the same job and is imho much more readable:
```
    if (repoName.endsWith(".git")) {
      return repoName.substring(repoName.length() - 4);
    }
    return repoName;
```

Reply:
yes it's somehow simpler but regex adds some more additional value (it checks that whole string is not `.git` or `something with .git`) providing some more insurance of errors in main regex

Reply:
My code I've wrote in a minute passed all your tests same as the regex I've tried to decipher for half an hour. Please write a tests for such cases you're talking about, then we can talk about what the code is providing.

I'm still skeptical you convince me that regex is better idea. However, I tend to have different opinions from others in team about clean and readable code. So if others thinks regex is better, I will not block it.

Reply:
I"ve addded `checkParsingBadRepositoryDoNotModifiesInitialInput` which passes both on master and regex version, but probably will fail on substring version.

Reply:
I see, for that cases I agree regex is more elegant solution as it's easier to list valid characters there. I still would like to avoid the grouping and lookahead pattern, because I think it adds unnecessary complexity. What about this:
```
if (repoName.matches("[\\w\\d._-]+.git")) {
  cut repoName
}
```
???

Reply:
Looks ok. Also, Sergey proposed solution when when non-matched value will fail immidiately throwing exception, which is also looks good since we can give a user his original input  https://github.com/eclipse/che/compare/master...skabashnyuk:gitFix?expand=1

Reply:
If we're talking about `7.9.1`, I'm against throwing an exception. It will change behavior of the method too much with potential side-effects. I'm for simplest and the least invasive fix now and let's talk about proper behavior of this class later, if needed.